ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,394 @@
+
+### Create Workflow Instance
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+WorkflowInstanceCreationIntent:Create | WorkflowInstanceCreationIntent:Created, Process:Element_Activating
+Process:Element_Activating | Process:Element_Activated, StartEvent:Element_Activating
+StartEvent:Element_Activating | StartEvent:Element_Activated, StartEvent:Element_Completing 
+StartEvent:Element_Completing | StartEvent:Element_Completed, Sequence_Flow_Taken, ServiceTask:Element_Activating 
+
+* the command `WorkflowInstanceCreationIntent:Create` is the initial command and start of the workflow instance
+* the event `Sequence_Flow_Taken` is produced by the command `StartEvent:Element_Completing` and not by an explicit command for the sequence flow (like `Take_Sequence_Flow`) because it can't be interrupted (i.e. no wait state)
+* since the command `StartEvent:Element_Completing` produces the event `Sequence_Flow_Taken`, it must produce the next command `ServiceTask:Element_Activating` to activate the target activity of the taken sequence flow
+
+### Service Task
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+ServiceTask:Element_Activating | ServiceTask:Element_Activated, Variable:Created, Job:Created
+JobBatch:Activate | JobBatch:Activated 
+Job:Complete | Job:Completed, ServiceTask:Element_Completing
+ServiceTask:Element_Completing | Variable: Created, ServiceTask:Element_Completed, Sequence_Flow_Taken, EndEvent:Element_Activating 
+
+* if the service task has an input variable mapping then it produces one or more `Variable:Created` events for the new local variables
+* the event `Job:Created` is produced by the command `ServiceTask:Element_Activating` instead of a `Job:Create` command because it can't be interrupted
+* if the job is completed with variables then these variables will be stored in the state until they are used for the output mapping (i.e. temporal/transient variables)
+* if the service task has an output variable mapping, or the job is completed with variables, then it produces one or more `Variable:Created`/`Variable:Updated` events for the new and updated variables of the workflow instance or scope
+
+### Message Intermediate Catch Event
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+IntermediateCatchEvent:Element_Activating | IntermediateCatchEvent:Element_Activated, MessageSubscription:Open
+MessageSubscription:Open | MessageSubscription:Opened, WorkflowInstanceSubscription:Open
+WorkflowInstanceSubscription:Open | WorkflowInstanceSubscription:Opened
+Message:Publish | Message:Published, WorkflowInstanceSubscription:Correlate
+WorkflowInstanceSubscription:Correlate | WorkflowInstanceSubscription:Correlated, MessageSubscription:Correlate, IntermediateCatchEvent:Element_Completing
+MessageSubscription:Correlate | MessageSubscription:Correlated
```

Comment:
This flow seems strange. Wouldn't both the MessageSubscription and the WorkflowInstanceSubscription have to be correlated before the IntermediateCatchEvent:Element_Completing follow-up command can be confidently written?